app/frontend/src/components/ui/history-panel.tsx

```diff
@@ -17,6 +18,30 @@ type Properties = {
 
 export default function HistoryPanel({ show, history, onClosed, onSelectedGroundingFile }: Properties) {
     const { t } = useTranslation();
+    const historyEndRef = useRef<HTMLDivElement>(null);
+
+    // Scroll to the bottom whenever the history changes
+    useEffect(() => {
+        if (historyEndRef.current) {
+            historyEndRef.current.scrollIntoView({ behavior: "smooth" });
+        }
+    }, [history]);
+
+    const formatTimestamp = (timestamp: Date) => {
```

Comment:
suggestion: date/time formats also need globalization, but I think this can be added later like it was done for localization.

Reply:
I made a change to handle globalization using the user's locale in my latest commit.